### PR TITLE
Feature // AutoFill after registration.

### PR DESCRIPTION
**Pull request by Mekhak11**
https://github.com/user-attachments/assets/9816c7df-61a1-447c-9b1d-1917147ae18c

### CODE REVIEW

File: Birthday/Birthday/Modules/Auth/ResetPassword/ViewModel/ResetPasswordViewModel.swift [hunk 1]
```diff
@@ -10,6 +10,7 @@ import Combine
 
 final class ResetPasswordViewModel: ResetPasswordViewModeling {
   
+  @Published var id: UUID = UUID()
```
Comment (Aslanyan99n) [inline Birthday/Birthday/Modules/Auth/ResetPassword/ViewModel/ResetPasswordViewModel.swift]: Please remove published for id

File: Birthday/Birthday/Modules/Auth/Landing/View/LandingScreen.swift [hunk 2]
```diff
@@ -10,14 +10,41 @@ import SwiftUI
...
+      case let .registration(viewModel):
+        hasher.combine(viewModel.id)
+      case let .forgotPassword(viewModel):
+        hasher.combine(viewModel.id)
+      case let .resetPassword(viewModel):
+        hasher.combine(viewModel.id)
+      }
+    }
+    
+    case signIn(viewModel: SignInViewModel)
```
Comment (Aslanyan99n) [inline Birthday/Birthday/Modules/Auth/Landing/View/LandingScreen.swift]: Please write cases top of enum

File: Birthday/Birthday/Modules/Auth/ResetPassword/ViewModel/ResetPasswordViewModel.swift (hunk 1 above)
Comment (Mekhak11) [reply Birthday/Birthday/Modules/Auth/ResetPassword/ViewModel/ResetPasswordViewModel.swift]: fixed.

File: Birthday/Birthday/Modules/Auth/Landing/View/LandingScreen.swift (hunk 2 above)
Comment (Mekhak11) [reply Birthday/Birthday/Modules/Auth/Landing/View/LandingScreen.swift]: fixed.

File: Birthday/Birthday/Modules/Auth/SignIn/ViewModel/SignInViewModel.swift [hunk 3]
```diff
@@ -33,8 +34,10 @@ class SignInViewModel: SignInViewModeling {
     email.isEmpty || password.isEmpty
   }
   
-  init(signInRepository: SignInRepository) {
+  init(signInRepository: SignInRepository, email: String = "", password: String = "") {
```
Comment (AkselAvetisyan) [inline Birthday/Birthday/Modules/Auth/SignIn/ViewModel/SignInViewModel.swift]: make this multiline for being more readable.

File: Birthday/Birthday/Modules/Auth/SignIn/ViewModel/SignInViewModel.swift [hunk 4]
```diff
@@ -25,6 +25,7 @@ class SignInViewModel: SignInViewModeling {
   
   var toastMessage: String = ""
   var isSuccessMessage: Bool = false
+  var id: UUID = UUID()
```
Comment (AkselAvetisyan) [inline Birthday/Birthday/Modules/Auth/SignIn/ViewModel/SignInViewModel.swift]: Change this var to let, also the other places

File: Birthday/Birthday/Modules/Auth/Landing/View/LandingScreen.swift [hunk 5]
```diff
@@ -73,14 +94,14 @@ extension LandingScreen {
...
-
+  
   private var signInButton: some View {
     LandingButton(
       title: String.Button.signIn,
       textColor: .rouge,
       backgroundColor: .bubblegumPink,
       action: {
-        authRouter.push(Screen.signIn)
+        authRouter.push(Screen.signIn(viewModel: SignInViewModel(signInRepository: SignInDefaultRepository())))
```
Comment (AkselAvetisyan) [inline Birthday/Birthday/Modules/Auth/Landing/View/LandingScreen.swift]: make multiline

File: Birthday/Birthday/Modules/Auth/SignIn/ViewModel/SignInViewModel.swift (hunk 4 above)
Comment (Mekhak11) [reply Birthday/Birthday/Modules/Auth/SignIn/ViewModel/SignInViewModel.swift]: fixed.

File: Birthday/Birthday/Modules/Auth/SignIn/ViewModel/SignInViewModel.swift (hunk 3 above)
Comment (Mekhak11) [reply Birthday/Birthday/Modules/Auth/SignIn/ViewModel/SignInViewModel.swift]: fixed.

File: Birthday/Birthday/Modules/Auth/Landing/View/LandingScreen.swift (hunk 5 above)
Comment (Mekhak11) [reply Birthday/Birthday/Modules/Auth/Landing/View/LandingScreen.swift]: fixed.

**Review by Aslanyan99n — DISMISSED** [review]

**Comment by AkselAvetisyan** [comment]
@Mekhak11 Why did you remove `id` from protocols?

**Review by nellyAslanyan — DISMISSED** [review]

**Review by AkselAvetisyan — APPROVED** [review]

**Pull request merged**